src/select.rs

```diff
@@ -196,6 +201,7 @@ impl<T: Display> Select<T> {
 
     fn handle_stop_filtering(&mut self, save: bool) {
         self.filtering = false;
+        self.cur_page = 0;
```

Comment:
we might want to reset the page to 1 since with an empty search result we can still navigate to a subsequent page